server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -395,7 +448,7 @@ private DiscoveryNode getPrimaryNode(ShardRouting primaryShard) {
     // visible to tests
     protected boolean processLatestReceivedCheckpoint(IndexShard replicaShard, Thread thread) {
         final ReplicationCheckpoint latestPublishedCheckpoint = latestReceivedCheckpoint.get(replicaShard.shardId());
-        if (latestPublishedCheckpoint != null && latestPublishedCheckpoint.isAheadOf(replicaShard.getLatestReplicationCheckpoint())) {
+        if (latestPublishedCheckpoint != null) {
```

Comment:
This is redundant and is covered inside of onNewCheckpoint's `shouldProcessCheckpoint`